bootstrap/kubeadm/api/v1alpha4/kubeadmconfig_types.go

```diff
@@ -94,6 +97,34 @@ type KubeadmConfigSpec struct {
 	// For more information, refer to https://github.com/kubernetes-sigs/cluster-api/pull/2763#discussion_r397306055.
 	// +optional
 	UseExperimentalRetryJoin bool `json:"useExperimentalRetryJoin,omitempty"`
+
+	// Ignition contains Ignition specific configuration.
+	//
+	// If set, bootstrap data will be generated in Ignition format instead of cloud-init.
```

Comment:
I think the comment is incorrect, bootstrap data will be generated in Igition format if the  `KubeadmConfigSpec.Format` == `Ignition`.

IIUC, it should be the additional or user defined ignition spec/config that will be "merged(*)" with the  ignition config genereated by bootstrapper controller.

see also below the suggestion of the name of the variables.

(*) What's the exactly merge policy in case of conflict? Who will have the priority? It will be  useful  if user defined config can override the config generated by the bootstrapper controller.

Reply:
In Ignition spec 3.x, fields from the child config (the one being merged) override corresponding fields in the parent config (the one specifying a config to be merged).  List elements from the child are generally merged field-by-field with the element from the parent that shares the same lookup key, or appended to the list if no such element exists.  Some lists (e.g. lists of command-line arguments) are special and always append rather than merging.  The rules are documented in more detail [here](https://coreos.github.io/ignition/operator-notes/#config-merging).

Spec 2.x behaves similarly, except that there is no list merging.  List elements from the child are appended to the corresponding list in the parent.  This can create e.g. duplicate file elements, in which case each element is applied in turn.

Reply:
>I think the comment is incorrect, bootstrap data will be generated in Igition format if the KubeadmConfigSpec.Format == Ignition.

Yes, this is outdated from original implementation. Removed.

>IIUC, it should be the additional or user defined ignition spec/config that will be "merged(*)" with the ignition config genereated by bootstrapper controller.

There is. In `Ignition.ContainerLinuxConfig.AdditionalConfig` field you can specify your own configuration which will be merged with bootstrap data generated by the controller. User-specified data takes precedence over controller data.

However, other Ignition frontends may behave differently.

Reply:
Ha, I missed that it was the "Ignition.ContainerLinuxConfig.AdditionalConfig" get "appended", not the `Ignition`. In that case, I 'm happy to keep using existing function name unchanged.